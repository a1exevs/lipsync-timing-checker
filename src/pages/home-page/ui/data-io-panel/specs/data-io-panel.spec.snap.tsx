import { render } from '@testing-library/react';
import React from 'react';

import DataIOPanel from 'src/pages/home-page/ui/data-io-panel/data-io-panel';

describe('DataIOPanel', () => {
  const defaultProps = {
    audioFileName: 'audio.mp3',
    wordsDataFileName: 'data.json',
    onAudioFileChange: jest.fn(),
    onWordsDataFileChange: jest.fn(),
    isLoadJSONDataButtonEnabled: true,
    isDownloadJSONDataButtonEnabled: true,
    onDownloadJSONDataClick: jest.fn(),
  };

  it('matches snapshot (default)', () => {
    const { container } = render(<DataIOPanel {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot (disabled buttons)', () => {
    const { container } = render(
      <DataIOPanel {...defaultProps} isLoadJSONDataButtonEnabled={false} isDownloadJSONDataButtonEnabled={false} />,
    );
    expect(container).toMatchSnapshot();
  });
});
