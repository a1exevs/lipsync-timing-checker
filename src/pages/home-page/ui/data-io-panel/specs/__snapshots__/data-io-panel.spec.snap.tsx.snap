// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DataIOPanel matches snapshot (default) 1`] = `
<div>
  <section
    class="flex flex-col md:flex-row md:justify-between items-center my-4 gap-4 rounded-md bg-gray-800 p-4 shadow"
  >
    <div
      class="flex flex-col gap-3 w-full md:w-1/2"
    >
      <div
        class="flex flex-row gap-2 items-center"
      >
        <button
          class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
          type="button"
        >
          <span>
            <svg
              aria-hidden="true"
              class="lucide lucide-file-audio"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.5 22h.5a2 2 0 0 0 2-2V7l-5-5H6a2 2 0 0 0-2 2v3"
              />
              <path
                d="M14 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M2 19a2 2 0 1 1 4 0v1a2 2 0 1 1-4 0v-4a6 6 0 0 1 12 0v4a2 2 0 1 1-4 0v-1a2 2 0 1 1 4 0"
              />
            </svg>
          </span>
          <span>
            Load audio
          </span>
        </button>
        <span
          class="flex-1 block truncate text-sm text-gray-300"
          title="audio.mp3"
        >
          audio.mp3
        </span>
        <input
          accept="audio/*"
          data-testid="file-input-audio"
          style="display: none;"
          type="file"
        />
      </div>
      <div
        class="flex flex-row gap-2 items-center"
      >
        <button
          class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
          type="button"
        >
          <span>
            <svg
              aria-hidden="true"
              class="lucide lucide-file-json"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"
              />
              <path
                d="M14 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M10 12a1 1 0 0 0-1 1v1a1 1 0 0 1-1 1 1 1 0 0 1 1 1v1a1 1 0 0 0 1 1"
              />
              <path
                d="M14 18a1 1 0 0 0 1-1v-1a1 1 0 0 1 1-1 1 1 0 0 1-1-1v-1a1 1 0 0 0-1-1"
              />
            </svg>
          </span>
          <span>
            Load JSON data
          </span>
        </button>
        <span
          class="flex-1 block truncate text-sm text-gray-300"
          title="data.json"
        >
          data.json
        </span>
        <input
          accept="application/json"
          data-testid="file-input-json"
          style="display: none;"
          type="file"
        />
      </div>
    </div>
    <button
      class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-red-600 hover:bg-red-700 text-white"
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="lucide lucide-download"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 15V3"
          />
          <path
            d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
          />
          <path
            d="m7 10 5 5 5-5"
          />
        </svg>
      </span>
      <span>
        Download JSON data
      </span>
    </button>
  </section>
</div>
`;

exports[`DataIOPanel matches snapshot (disabled buttons) 1`] = `
<div>
  <section
    class="flex flex-col md:flex-row md:justify-between items-center my-4 gap-4 rounded-md bg-gray-800 p-4 shadow"
  >
    <div
      class="flex flex-col gap-3 w-full md:w-1/2"
    >
      <div
        class="flex flex-row gap-2 items-center"
      >
        <button
          class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
          type="button"
        >
          <span>
            <svg
              aria-hidden="true"
              class="lucide lucide-file-audio"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.5 22h.5a2 2 0 0 0 2-2V7l-5-5H6a2 2 0 0 0-2 2v3"
              />
              <path
                d="M14 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M2 19a2 2 0 1 1 4 0v1a2 2 0 1 1-4 0v-4a6 6 0 0 1 12 0v4a2 2 0 1 1-4 0v-1a2 2 0 1 1 4 0"
              />
            </svg>
          </span>
          <span>
            Load audio
          </span>
        </button>
        <span
          class="flex-1 block truncate text-sm text-gray-300"
          title="audio.mp3"
        >
          audio.mp3
        </span>
        <input
          accept="audio/*"
          data-testid="file-input-audio"
          style="display: none;"
          type="file"
        />
      </div>
      <div
        class="flex flex-row gap-2 items-center"
      >
        <button
          class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-blue-600 hover:bg-blue-700 text-white w-1/2"
          disabled=""
          type="button"
        >
          <span>
            <svg
              aria-hidden="true"
              class="lucide lucide-file-json"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V7Z"
              />
              <path
                d="M14 2v4a2 2 0 0 0 2 2h4"
              />
              <path
                d="M10 12a1 1 0 0 0-1 1v1a1 1 0 0 1-1 1 1 1 0 0 1 1 1v1a1 1 0 0 0 1 1"
              />
              <path
                d="M14 18a1 1 0 0 0 1-1v-1a1 1 0 0 1 1-1 1 1 0 0 1-1-1v-1a1 1 0 0 0-1-1"
              />
            </svg>
          </span>
          <span>
            Load JSON data
          </span>
        </button>
        <span
          class="flex-1 block truncate text-sm text-gray-300"
          title="data.json"
        >
          data.json
        </span>
        <input
          accept="application/json"
          data-testid="file-input-json"
          style="display: none;"
          type="file"
        />
      </div>
    </div>
    <button
      class="inline-flex items-center justify-center gap-2 rounded-sm px-4 py-2 text-sm font-medium transition disabled:cursor-not-allowed disabled:opacity-50 bg-red-600 hover:bg-red-700 text-white"
      disabled=""
      type="button"
    >
      <span>
        <svg
          aria-hidden="true"
          class="lucide lucide-download"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 15V3"
          />
          <path
            d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
          />
          <path
            d="m7 10 5 5 5-5"
          />
        </svg>
      </span>
      <span>
        Download JSON data
      </span>
    </button>
  </section>
</div>
`;
